mobile/src/test/java/org/openhab/habdroid/model/OpenHAB2WidgetTest.java

```diff
@@ -1,67 +1,404 @@
-                "  }";
+    public void testGetIconPath() throws Exception {
+        assertEquals("icon/groupicon?state=OFF&format=PNG", sut1.getIconPath());
+        assertEquals("icon/groupicon?state=ON&format=SVG", sut2.getIconPath());
+        assertEquals("icon/slider?state=81&format=SVG", sut3.getChildren().get(0).getIconPath());
+
+        // Rollersutter icon should always be 0 to 100, not ON/OFF
+        //assertEquals("icon/rollershutter?state=0&format=SVG", sut3.getChildren().get(1).getIconPath());
+        assertEquals("icon/rollershutter?state=42&format=SVG", sut3.getChildren().get(2).getIconPath());
+
+    }
+
+    @Test
+    public void testGetChildren() throws Exception {
+        assertEquals("demo11", sut1.getChildren().get(0).getId());
+
+        // sut2 doesn't have children
+        IndexOutOfBoundsException e = null;
+        try {
+            sut2.getChildren().get(0).getId();
+        } catch (IndexOutOfBoundsException ex) {
+            e = ex;
+        }
+        assertTrue(e != null);
+    }
+
+    @Test
+    public void testHasChildren() throws Exception {
+        assertEquals(true, sut1.hasChildren());
+        assertEquals(false, sut2.hasChildren());
+        assertEquals(true, sut3.hasChildren());
+    }
+
+    @Test
+    public void testGetPeriode() throws Exception {
+        assertEquals("M", sut1.getPeriod());
+        assertEquals("D", sut2.getPeriod()); // Object has no period, should default to D
+    }
+
+    @Test
+    public void testGetStep() throws Exception {
+        assertEquals(1F, sut1.getStep());
+        assertEquals(0.1F, sut2.getStep());
+    }
+
+    @Test
+    public void testGetRefresh() throws Exception {
+        assertEquals(1000, sut1.getRefresh());
+        assertEquals(100, sut2.getRefresh()); // Min refresh is 100, object has set refresh to 10
+    }
+
+    @Test
+    public void testHasItem() throws Exception {
+        assertEquals(true, sut1.hasItem());
+        assertEquals(true, sut2.hasItem());
+    }
+
+    @Test
+    public void testHasLinkedPage() throws Exception {
+        assertEquals(true, sut1.hasLinkedPage());
+        assertEquals(false, sut2.hasLinkedPage());
+    }
+
+    @Test
+    public void testHasMappings() throws Exception {
+        assertEquals(true, sut1.hasMappings());
+        assertEquals(true, sut2.hasMappings());
+    }
+
+    @Test
+    public void testChildrenHasLinkedPages() throws Exception {
+        assertEquals(false, sut1.childrenHasLinkedPages());
+        assertEquals(false, sut2.childrenHasLinkedPages());
+        assertEquals(true, sut3.childrenHasLinkedPages());
+
+    }
+
+    @Test
+    public void testChildrenHasNonlinkedPages() throws Exception {
+        assertEquals(true, sut1.childrenHasNonlinkedPages());
+        assertEquals(false, sut2.childrenHasNonlinkedPages());
+
+    }
+
+    @Test
+    public void testGetColors() throws Exception {
+        // Doesn't seem to work via Android Studio
```

Comment:
add:
```
import android.graphics.Color;
import static org.hamcrest.CoreMatchers.equalTo;
// ...
Assume.assumeThat(Color.parseColor("blue"), equalTo(Color.BLUE));
```

To the beginning of the task and the comment above. This will mark the test as skipped, so a build does not fail, but we know, that there's something not working :)